src/main/java/detail/JStatGuiGlobalData.java

```diff
@@ -0,0 +1,17 @@
+package detail;
+
+import java.util.concurrent.ExecutorService;
+
+public class JStatGuiGlobalData {
+
+    /**
+     * The container used to hold the loaded data sets
+     */
+    public static IDataSetContainer dataSetContainer;
+
+
+    /**
+     * A set of worker pools used around
+     */
+    public static ExecutorService workersPool;
```

Comment:
can you pls explain what is executor service is going to do?

Reply:
This is a pool of threads meant to do the heavy lifting...e.g. load a dataset, perform a calculation. checkout  the class ```LoadDataSet``` and you will see what I mean.